cluster-autoscaler/simulator/scheduler_based_predicates_checker.go

```diff
@@ -72,6 +72,17 @@ func NewSchedulerBasedPredicateChecker(kubeClient kube_client.Interface, stop <-
 	// informerFactory....Lister()/informerFactory....Informer() methods
 	informerFactory.Start(stop)
 
+	// Also wait for all informers to be up-to-date. This is necessary for
+	// objects that were added when creating the fake client. Without
+	// this wait, those objects won't be visible via the informers when
+	// the test runs.
+	synced := informerFactory.WaitForCacheSync(stop)
```

Comment:
I'm worried about this without timeout. I've seen cases of WaitForCacheSync() hanging ~forever.